### PR TITLE
Doc: Improve readme

### PR DESCRIPTION
**Pull request by khaeljy**
Time spent on this PR: 30m

Resolves: N/A

# Pull Request type

Please check the type of change your PR introduces:

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, renaming)
-   [ ] Refactoring (no functional changes, no API changes)
-   [ ] Build-related changes
-   [x] Documentation content changes
-   [ ] Testing

# What is the new behavior?

I've added a bit of context to the README to help new contributors (like me).

# Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 This repository contains the execution of the EF standard execution layer tests.
 
+The Ethereum Foundation provides a suite of [official tests](https://github.com/ethereum/tests) to verify the compliance of EVM clients.
+Passing all these tests qualifies a client as EVM-compliant.
```
Comment (greged93) [inline README.md:6]:
```suggestion
Passing all these tests qualifies allows a client to gain confidence on his execution layer.
```

File: README.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
+For further information, please refer to the [official documentation](https://ethereum-tests.readthedocs.io/en/latest/).
+
+As Kakarot is an EVM running within CairoVM, we can't run these tests using the Ethereum Foundation runner ([retesteth](https://github.com/ethereum/retesteth)).
+We therefore need to develop our own runner to be able to run these tests on Kakarot and thus certify our compatibility with the EVM.
```
Comment (greged93) [inline README.md:10]:
```suggestion
Kakarot is an EVM running within CairoVM, coupled with a [RPC](https://github.com/kkrt-labs/kakarot-rpc/tree/main), which would make it possible to run these tests using the Ethereum Foundation runner ([retesteth](https://github.com/ethereum/retesteth)). However, in order to limit the possible number of interactions and avoid adding failing points, we develop our own simplified test runner based on Reth's ef-tests runner.
```

**Review by greged93 — COMMENTED** [review]
just a couple of clarifications, feel free to let me know if you agree, otherwise thanks for the PR!

**Review by Eikix — APPROVED** [review]

**Comment by Eikix** [comment]
thanks for the PR!:)

**Comment by khaeljy** [comment]
> just a couple of clarifications, feel free to let me know if you agree, otherwise thanks for the PR!

thank you for these details 😉

**Review by Eikix — APPROVED** [review]

**Pull request merged**